### PR TITLE
Logic feature

### PR DESCRIPTION
**Pull request by Oplia512**
JsonParser and tests

### CODE REVIEW

File: src/main/java/com/java_academy/logic/jsonModel/JsonMessage.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.java_academy.logic.jsonModel;
+
+/**
+ * Interface for json message in application. It is for determine type of message.
+ * */
+public interface JsonMessage {
```
Comment (mhbratek) [inline src/main/java/com/java_academy/logic/jsonModel/JsonMessage.java:6]: Missing JavaDoc.

File: src/main/java/com/java_academy/logic/model/Players.java [hunk 2]
```diff
@@ -6,7 +6,7 @@
     FIRST_PLAYER(new Player("First", 3)) {
         @Override
         public Players getOpponent() {
-            return FIRST_PLAYER;
+            return Players.SECOND_PLAYER;
```
Comment (mhbratek) [inline src/main/java/com/java_academy/logic/model/Players.java:9]: :+1:

File: src/main/java/com/java_academy/logic/state_machine/PlayerEndActionState.java [hunk 3]
```diff
@@ -13,7 +13,7 @@
 public class PlayerEndActionState implements GameState {
 
     private Players currentPlayer;
-    private Boolean somethingHitted;
+    private Boolean hasBeenHit;
```
Comment (mhbratek) [inline src/main/java/com/java_academy/logic/state_machine/PlayerEndActionState.java:16]: Nice name.

File: src/test/java/com/java_academy/logic/model/PlayerTest.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	private int nukeCounter = 3;
+
+	@BeforeTest
+	public void createPlayer() {
+		player = new Player("first", nukeCounter);
+		player.createFleet();
+	}
+	
+	@Test
+	public void testPlayerCanUseNuke() {
```
Comment (mhbratek) [inline src/test/java/com/java_academy/logic/model/PlayerTest.java:24]: Think about change name convention. for example 
shouldReturnTrueIfPlayerCanUseNuke

**Review by mhbratek — APPROVED** [review]

**Review by pawlusha — APPROVED** [review]

**Pull request merged**